include/aikido/planner/ompl/detail/OMPLConfigurationToConfigurationPlanner-impl.hpp

```diff
@@ -0,0 +1,207 @@
+#ifndef AIKIDO_PLANNER_OMPL_DETAIL_OMPLCONFIGURATIONTOCONFIGURATION_IMPL_HPP_
+#define AIKIDO_PLANNER_OMPL_DETAIL_OMPLCONFIGURATIONTOCONFIGURATION_IMPL_HPP_
+
+#include "aikido/planner/ompl/OMPLConfigurationToConfigurationPlanner.hpp"
+
+#include <utility>
+
+#include "aikido/constraint/TestableIntersection.hpp"
+#include "aikido/constraint/dart/FrameDifferentiable.hpp"
+#include "aikido/constraint/dart/FrameTestable.hpp"
+#include "aikido/constraint/dart/JointStateSpaceHelpers.hpp"
+#include "aikido/distance/defaults.hpp"
+#include "aikido/planner/ompl/GeometricStateSpace.hpp"
+#include "aikido/planner/ompl/MotionValidator.hpp"
+#include "aikido/planner/ompl/Planner.hpp"
+#include "aikido/statespace/GeodesicInterpolator.hpp"
+
+namespace aikido {
+namespace planner {
+namespace ompl {
+
+using aikido::constraint::dart::createProjectableBounds;
+using aikido::constraint::dart::createSampleableBounds;
+using aikido::constraint::dart::createTestableBounds;
+using aikido::constraint::dart::createProjectableBounds;
+using aikido::distance::createDistanceMetric;
+using aikido::statespace::dart::MetaSkeletonStateSpace;
+
+//==============================================================================
+template <class PlannerType>
+OMPLConfigurationToConfigurationPlanner<PlannerType>::
+    OMPLConfigurationToConfigurationPlanner(
+        statespace::ConstStateSpacePtr stateSpace,
+        common::RNG* rng,
+        statespace::ConstInterpolatorPtr interpolator,
+        distance::DistanceMetricPtr dmetric,
+        constraint::SampleablePtr sampler,
+        constraint::TestablePtr boundsConstraint,
+        constraint::ProjectablePtr boundsProjector,
+        double maxDistanceBtwValidityChecks)
+  : ConfigurationToConfigurationPlanner(std::move(stateSpace), rng)
+  , mInterpolator(std::move(interpolator))
+  , mBoundsConstraint(std::move(boundsConstraint))
+  , mMaxDistanceBtwValidityChecks(maxDistanceBtwValidityChecks)
+{
+  if (!mInterpolator)
+    mInterpolator = std::make_shared<aikido::statespace::GeodesicInterpolator>(
+        mStateSpace);
+
+  if (!dmetric)
+    dmetric = std::move(createDistanceMetric(mStateSpace));
+
+  // TODO (avk): The constraint namespace functions only work with
+  // metaskeletonstatespace
+  const auto metaskeletonStatespace
+      = std::dynamic_pointer_cast<const MetaSkeletonStateSpace>(mStateSpace);
+
+  if (!metaskeletonStatespace
```

Comment:
I found this logic somewhat confusing because it wasn't clear to me when all of the conditions would be satisfied. I think this might be clearer:
```cpp
if (metaSkeletonStateSpace)
{
  if (!sampler) ...
  if (!mBoundsConstraint) ...
  if (!boundsProjector) ...
}
else
{
  if (!sampler || !boundsConstraint || !boundsProjector)
  {
    throw ...
  }
}

auto sspace = ...
```

Reply:
sigh